src/vm/syncblk.cpp

```diff
@@ -3049,6 +3051,7 @@ BOOL AwareLock::EnterEpilogHelper(Thread* pCurThread, INT32 timeOut)
     FireEtwContentionStart_V1(ETW::ContentionLog::ContentionStructs::ManagedContention, GetClrInstanceId());
 
     LogContention();
+    InterlockedIncrement64((LONGLONG *)&s_lockContentionCount);
```

Comment:
Isn't this going to make contention even more expensive, creating contention not only on this individual monitor but synchronizing it with every other monitor experiencing contention as well, at least a little?

Reply:
This thread will have to wait for the lock to be released, but the increment itself may not contend with other threads. It can add a little bit of overhead but we're about to spend several microseconds for context switching anyway so it should be negligible. It is still extra work though and this can add a little bit of CPU time in cases of frequent contention.

Reply:
All cores will start fighting over this cache line if there is a bit contention (that can be even spread over multiple locks). Each of these fights can cost 1000s cycles on many core systems, and the amplify the effect of contention that would be harmless otherwise.

I would recommend testing this on server class hardware with 32+ cores.

Reply:
CoreCLR is using thread-locals for tracking thread pool work item completion counts, maybe that is the reason as well and all of those should be switched. Will try it out on a larger machine.

Reply:
See latest comment about perf, I didn't notice a perf difference but I have switched to use thread-locals